Makefile

```
@@ -0,0 +1,20 @@
+PREFIX ?= /usr/local
+BASH_COMPLETION_DIR := $(shell pkg-config --variable=completionsdir bash-completion) 
+# pkg-config adds a space for some reason, we have to strip it off.
+BASH_COMPLETION_DIR := $(strip $(BASH_COMPLETION_DIR))
+USERDIR ?= $(HOME)
+
+install: 
+	@if [ -d $(BASH_COMPLETION_DIR) ]; then \
+	 	echo $(BASH_COMPLETION_DIR);\
+		cp notes.bash_completion "$(BASH_COMPLETION_DIR)/notes";\
+	fi # Small test for bash completion support
+	install -m755 -D notes $(PREFIX)/bin/notes
+	install -m777 -D config.example $(USERDIR)/.config/notes/config.example
+	install -D notes.1 $(PREFIX)/share/man/man1/notes.1
+	mandb 1>/dev/null
```

On osx, this line fails, as there's no mandb. Doesn't seem to be necessary though - `man notes` works fine after installation. Can we make this optional, and silently skip it if `mandb` doesn't exist?

---

We could do with more error handling here too, since this could easily silently go wrong. We should `set -o errexit`, for starters, and trap the error to display a nice message like, along the lines of:

> Oh no, notes failed to install! Please follow the manual install steps at github.com/pimterry/notes, and file a bug (if one doesn't already exist) with the issue you're seeing, so we can get it fixed for you.

This bit could be a separate issue actually, doesn't need to block this. Feel free to file one and do those changes later if you'd prefer.

---

Some success output generally is probably a good idea too actually, to explain what's just happened and get people started:

> Notes has been installed to $(PREFIX)/bin/notes. Config has been created in $(exampleconfigpath), which you can edit if you'd like to change the default settings.
> Get started now by running `notes open my-new-note`, or `notes help` for more info.

---

> man notes works fine after installation. Can we make this optional, and silently skip it if mandb doesn't exist?

We can probably get away without mandb then. Thoughts?